api/v1/tenant_types.go

```diff
@@ -24,6 +24,25 @@ type TenantSpec struct {
 	// ID is the unique identifier for this tenant.
 	// +required
 	ID string `json:"id,omitempty"`
+
+	// DataRetention defines the how to store a tenant's data
+	DataRetention []DataRetention `json:"dataRetention"`
```

Comment:
I think `DataRetrention` is no longer the right name for this - it really is just configuring indices. Maybe just call the field `Indices`? 

I think we should set validation on this field that ensures:
- Each element is unique, and there are no duplicates in the array.
- If any indices are specified, then _all_ indices should be specified. I don't think it's valid to just include one index here, and it will be easy to accidentally forget one.

Reply:
A good test that shows one way in which the naming isn't quite right - `DataRetention.Type` implies "What type of data retention?" but actually the field means "What type of data does this refer to"

Reply:
I think we can do the uniqueness check via kubebuilder (but not 100% sure).

Otherwise, it should probably be the Linseed controller for now, given that's where this field is used.

Reply:
I did try the kubebuilder tag, but then I saw FV failures: (https://tigera.semaphoreci.com/jobs/26a53ce8-5407-481d-aed2-8150f322bfd9)

```
customresourcedefinition.apiextensions.k8s.io/tigerastatuses.operator.tigera.io created 01:51
The CustomResourceDefinition "tenants.operator.tigera.io" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[indices].uniqueItems: Forbidden: uniqueItems cannot be set to true since the runtime complexity becomes quadratic 01:51
Makefile:403: recipe for target 'deploy-crds' failed
```